### PR TITLE
Reduce the scope of Maven builds

### PR DESCRIPTION
**Pull request by msridhar**
Maven builds now only do close to the minimum necessary to run the `com.ibm.wala.ide.*` Eclipse plug-in tests.  Unfortunately we can't easily port these test runs to Gradle as they rely on [Tycho](https://www.eclipse.org/tycho/), which is Maven-only.

As part of this change we do the following:
1. Remove most sub-modules from the top-level `pom.xml` so Maven no longer builds them.
2. Change the `com.ibm.wala.ide.*` projects to rely on the other WALA projects as Maven artifacts.  This means that `./gradlew install` must be run first for a Maven build to succeed or to see updated versions of other WALA code.  (Maybe we should run this as part of the Maven build?)
3. Related to 2, update the relevant Eclipse plug-in metadata in `com.ibm.wala.ide.*`.  In particular, we need exactly one plug-in to manage each core WALA Maven artifact jar and re-export its relevant packages.  Probably more packages are exported than needed right now; we can trim it down later if needed (I doubt it matters).
4. Update relevant Travis CI scripts in light of 2.

With these changes, a Maven build on Travis CI with a warm cache runs in under 5 minutes.

Note that we still pull in test data for `com.ibm.wala.ide.jdt.test` in a manner outside the WALA Maven artifacts, due to the particular structure required.  (`wala.cast.java.test.data` must be zipped in the structure of an Eclipse project and that zip needs to be copied over.)  We can further clean this up in the future.

This change is an intermediate step toward #462 and retiring Maven, and reduces the scope of Maven usage, so it is worth landing on its own.  A follow-up PR will remove various Maven, Ant, and Eclipse build configuration files that are no longer needed.

### CODE REVIEW

File: com.ibm.wala.cast.java.test.data/build.xml [hunk 1]
```diff
@@ -78,8 +68,8 @@
 	   excludes="**/Activator.java"/>
   </target>
   
-  <target name="jar" depends="compile,testdatadir">
-    <jar destfile="${basedir}/../com.ibm.wala.ide.jdt.test/testdata/test_project.zip" manifest="${basedir}/META-INF/MANIFEST.MF" basedir="${basedir}"/>
+  <target name="jar" depends="compile">
```
Comment (msridhar) [inline com.ibm.wala.cast.java.test.data/build.xml]: I suspect this was the main reason we had to run `mvn verify` twice to get the tests to pass before.  This script was writing a zip file directly to `com.ibm.wala.ide.jdt.test` but that project declared no dependence on this one.  I've fix things so the zip file gets written locally and then copied over by `com.ibm.wala.ide.jdt.test`, and that a dependence exists in the build.

File: travis/before-install-maven [hunk 2]
```diff
@@ -1,8 +1,6 @@
 #!/bin/sh -eux
 
-./gradlew prepareMavenBuild
-
-git clone --depth=1 https://github.com/secure-software-engineering/DroidBench.git /tmp/DroidBench
+./gradlew prepareMavenBuild install -x javadoc
```
Comment (liblit) [inline travis/before-install-maven]: Ah, nice idea to exclude `javadoc` here!

Comment (msridhar) [reply travis/before-install-maven]: Yes, I also exclude javadoc when running the main Maven build.  We generate javadoc for everything during the Gradle build, so no need to do it here.  And it's surprising how expensive javadoc generation is!  It takes longer than compilation.

**Comment by msridhar** [comment]
My understanding of the Eclipse build system is also not deep (though it's deeper than it was a few days ago...).  I was highly reliant on keeping the regression tests passing for sanity-checking my changes.  Since these changes succeed in doing that, and we have few to no users of the IDE code (as far as we know), this is good enough I think.

**Pull request merged**